internal/provider/kubernetes/controller.go

```diff
@@ -1273,18 +1273,33 @@ func (r *gatewayAPIReconciler) processBackendTLSPolicies(
 }
 
 // processBackends adds Backends to the resourceTree
-func (r *gatewayAPIReconciler) processBackends(ctx context.Context, resourceTree *resource.Resources) error {
+func (r *gatewayAPIReconciler) processBackends(ctx context.Context, resourceTree *resource.Resources, resourceMappings *resourceMappings) error {
 	backends := egv1a1.BackendList{}
 	if err := r.client.List(ctx, &backends); err != nil {
 		return fmt.Errorf("error listing Backends: %w", err)
 	}
 
 	for _, backend := range backends.Items {
 		backend := backend //nolint:copyloopvar
+		// We only interested in Backends that are referenced by GatewayClass.
+		if !resourceMappings.allAssociatedBackendRefs.Has(gwapiv1.BackendObjectReference{
```

Comment:
imo this  `!resourceMappings.allAssociatedBackendRefs.Has` is helping us skip 2 lines, and we are calling this again right after, should we avoid it here to improve readibility ?

Reply:
so this is for not adding backends not relevant to the currently-processed GatewayClass to `resourceTree.Backends`.

vs

Below is to avoid one Backend being appended twice to `resourceTree.Backends` as it can be tried multiple times later in `processBackendRef`.  I think yeah if-block below can be removed agree but both logics are necessary